rotkehlchen/tests/exchanges/test_bitfinex.py

```diff
@@ -0,0 +1,1355 @@
+import warnings as test_warnings
+from contextlib import ExitStack
+from datetime import datetime
+from http import HTTPStatus
+from unittest.mock import call, patch
+
+import pytest
+from requests import Response
+
+from rotkehlchen.accounting.structures import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.assets.converters import (
+    BITFINEX_EXCHANGE_TEST_ASSETS,
+    UNSUPPORTED_BITFINEX_ASSETS,
+)
+from rotkehlchen.errors import (
+    RemoteError,
+    SystemClockNotSyncedError,
+    UnknownAsset,
+    UnsupportedAsset,
+)
+from rotkehlchen.exchanges.bitfinex import (
+    API_KEY_ERROR_CODE,
+    API_KEY_ERROR_MESSAGE,
+    API_RATE_LIMITS_ERROR_MESSAGE,
+    API_SYSTEM_CLOCK_NOT_SYNCED_ERROR_CODE,
+    Bitfinex,
+    CurrencyMapResponse,
+)
+from rotkehlchen.exchanges.data_structures import AssetMovement, Trade, TradeType
+from rotkehlchen.fval import FVal
+from rotkehlchen.tests.utils.mock import MockResponse
+from rotkehlchen.typing import (
+    AssetAmount,
+    AssetMovementCategory,
+    Fee,
+    Location,
+    Price,
+    Timestamp,
+    TradePair,
+)
+
+
+def test_name():
+    exchange = Bitfinex('a', b'a', object(), object())
+    assert exchange.name == str(Location.BITFINEX)
+
+
+def test_bitfinex_exchange_assets_are_known(mock_bitfinex):
```

Comment:
This logic is driven by the exchange pairs. This test won't log anything given the following case:
- Rotki does not support asset X but Bitfinex does, happens that not on any exchange pair (e.g. margin or futures).

I assume this is the expected behaviour.

Reply:
Hmmm but that's bad. We want these tests exactly because we want to detect this case.

Unless you mean that we have it in the `UNSUPPORTED_ASSETS` for bitfinex already. If that's the case it's fine since we already know it's not supported.